3scale/porta
# THREESCALE-8722: Add low priority Sphinx indexation worker

**What this PR does / why we need it**:

Executing a full reindex sends the Sphinx indexation jobs to the "default" sidekiq queue. This may delay the processing of some of the other jobs that are also using the "default" queue, and also can almost block the lower-priority queues.

These PR facilitates moving the reindexing jobs into a separate queue, with lower priority. This would give us two main benefits:

- the reindexing jobs will be executed with lower priority so will not affect other more time-critical jobs
- it will be possible to delete all reindexing jobs in case something goes wrong

The queue name is configured via a property `bulk_indexing_queue` in the `settings.yml` file. If the value is not set, the indexing jobs will be queued in the `default` queue.

**NOTE:** this only affects the reindexing triggered by the rake task `sphinx:enqueue`. The indexing jobs that are enqueued automatically on models changes keep being sent to the `default` queue.

**Which issue(s) this PR fixes** 

Implements https://issues.redhat.com/browse/THREESCALE-8722

**Verification steps** 

1. Verify custom queue:
> set `bulk_indexing_queue` property in `settings.yml` to `bulk_indexing`
> run `bundle exec rake sphinx:enqueue` and ensure that the jobs are enqueued in the `bulk_indexing` queue
> make some changes on the different indexed models, and make sure that they are enqueued in the `default` queue

2. Verify default queue:
> remove `bulk_indexing_queue` property in `settings.yml`
> run `bundle exec rake sphinx:enqueue` and ensure that the jobs are enqueued in the `default` queue

**Special notes for your reviewer**:

This requires a change in the deployment, to add a new queue and set proper weight, see https://github.com/mperham/sidekiq/wiki/Advanced-Options#queues
